alibi_detect/utils/pytorch/kernels.py

```diff
@@ -29,12 +36,76 @@ def sigma_median(x: torch.Tensor, y: torch.Tensor, dist: torch.Tensor) -> torch.
     return sigma
 
 
-class GaussianRBF(nn.Module):
+class BaseKernel(nn.Module):
+    """
+    The base class for all kernels.
+    """
+    def __init__(self) -> None:
+        super().__init__()
+        self.parameter_dict: dict = {}
+        self.active_dims: Optional[list] = None
+        self.feature_axis: int = -1
```

Comment:
As discussed on call, I would expose this behaviour via a wrapper that handles the subsetting and feeding of inputs into a desired underlying kernel. Otherwise we'll have to duplicate this logic within every future kernel, as will users wishing to define their own.

Reply:
This is now done with a `DimensionSelectKernel`. So no selection is required for each kernel.

Reply:
On the one hand I like the flexibility of the new `DimensionSelectKernel`, but on the other it leads to quite a bit of verbosity: 
![image](https://user-images.githubusercontent.com/13080878/185423677-8ec7809f-42e8-4268-b160-d56bf107eacb.png)

Not sure if worth investigating other options here? E.g. the selection logic could be implemented as a `torch/tf` specific method in e.g. `TorchBaseKernel` and then it's application could be controlled via a parameter (could be the same `active_dims`) to `forward/call` - then custom kernels would be expected both to subclass but also if they want the selection capability to call `super().forward(x, y, active_dims)` to get the selected dimensions before applying custom logic.  This is just one idea, perhaps there are others. @arnaudvl @ojcobb